compiler/src/iree/compiler/InputConversion/MHLO/ConvertMHLOToLinalgExt.cpp

```diff
@@ -470,6 +471,99 @@ struct ReverseOpConversion : public OpConversionPattern<mhlo::ReverseOp> {
   }
 };
 
+//===----------------------------------------------------------------------===//
+// TopkOp
+//===----------------------------------------------------------------------===//
+
+struct TopkOpConversion : public OpConversionPattern<chlo::TopKOp> {
+  using OpConversionPattern<chlo::TopKOp>::OpConversionPattern;
+  LogicalResult matchAndRewrite(
+      chlo::TopKOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const final {
+    Location loc = op.getLoc();
+    Value operand = adaptor.operand();
+
+    auto inputValuesType = operand.getType().dyn_cast<ShapedType>();
+    auto outputValuesType = op.values().getType().dyn_cast<ShapedType>();
+    auto outputIndicesType = op.indices().getType().dyn_cast<ShapedType>();
+    if (!inputValuesType || !outputValuesType || !outputIndicesType) {
+      return rewriter.notifyMatchFailure(
+          op, "Input and output must be of ShapedType");
+    }
+
+    Type valueElementType = outputValuesType.getElementType();
+    Type indicesElementType = outputIndicesType.getElementType();
+    // Only handle integer types for indicies. Index type is not supported.
+    if (!indicesElementType.isa<IntegerType>()) {
+      return rewriter.notifyMatchFailure(
+          op, "Output indices must be of integer type.");
+    }
+
+    // Create and initialize output tensors for LinalgExt TopK results
+    // Define the output types based on the results of CHLO TopK
+    SmallVector<Value> dynSizes;
+    for (auto en : llvm::enumerate(inputValuesType.getShape())) {
+      if (en.value() == ShapedType::kDynamicSize) {
+        dynSizes.push_back(
+            rewriter.create<tensor::DimOp>(loc, adaptor.operand(), en.index()));
+      }
+    }
+    Value initTensorOutputValues = rewriter.create<mlir::linalg::InitTensorOp>(
+        loc, dynSizes, outputValuesType.getShape(), valueElementType);
+    Value initTensorOutputIndices = rewriter.create<mlir::linalg::InitTensorOp>(
+        loc, dynSizes, outputIndicesType.getShape(), indicesElementType);
+    // Initialize indices to 0 and values to negative infinity
+    Attribute negInfAttr;
+    if (auto intType = valueElementType.dyn_cast<IntegerType>()) {
+      negInfAttr = rewriter.getIntegerAttr(
+          intType, APInt::getSignedMinValue(intType.getWidth()));
+    } else {
+      auto negApFloat = APFloat::getInf(
+          valueElementType.cast<FloatType>().getFloatSemantics(),
+          /*Negative=*/true);
+      negInfAttr = rewriter.getFloatAttr(valueElementType, negApFloat);
+    }
+    Value negInf = rewriter.create<arith::ConstantOp>(loc, negInfAttr);
+    Attribute posInfAttr = rewriter.getIntegerAttr(
+        indicesElementType, APInt::getSignedMaxValue(32));
+    Value posInf = rewriter.create<arith::ConstantOp>(loc, posInfAttr);
```

Comment:
If I am reading this correctly (and following Hanhans comments), we should not need to initialize the indices tensor, unless this is some runtime error mode that is being signalled. This would allow the `fill` op for the output indices to be dropped.